hawkmoth/ext/transformations/__init__.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2023, Jani Nikula <jani@nikula.org>
+# Licensed under the terms of BSD 2-Clause, see LICENSE for details.
+
+def _process_docstring(app, lines, transform, options):
+    transformations = app.config.cautodoc_transformations
+    tropt = options.get('transform')
+
+    if transformations is None:
+        return
+
+    # Note: None is a valid key for default.
+    if tropt not in transformations:
+        return
+
+    # Note: None is a valid value for no transformation.
+    transform = transformations.get(tropt)
+    if transform is None:
+        return
+
+    comment = '\n'.join(lines)
+    comment = transform(comment)
+    lines[:] = comment.splitlines()[:]
+
+def setup(app):
```

Comment:
I'd consider adding deprecation notes. This might be new code, but it's an old feature we want to get rid of.

Reply:
I just get an uneasy feeling forcing a number of consecutive compat/transform changes on users...

Reply:
Do we have users? :sweat_smile: 

In any case, the warnings should be gated on actual use of these functions, them being there doesn't mean we're about to remove them next week and we haven't released v1... We can delay the deprecation though, it was only a suggestion.

Reply:
Cleaned up the documentation text in #131, but opted not to deprecate this yet. The documentation implies this is going away in the future.